### PR TITLE
docs: document hook properties

### PR DESCRIPTION
**Pull request by wadjih-bencheikh18**
closes : https://github.com/zakodium-oss/react-plot/issues/415

### CODE REVIEW

File: src/hooks/useAxisWheelZoom.tsx [hunk 1]
```diff
@@ -6,10 +6,31 @@ import {
 import { ControllerHookOptions } from './types';
 
 export interface UseAxisWheelZoomOptions extends ControllerHookOptions {
+  /**
+   * OnWheel zoom direction
+   * @default 'vertical'
+   * */
   direction?: 'horizontal' | 'vertical';
+  /**
+   * Zoom axisId
```
Comment (stropitek) [inline src/hooks/useAxisWheelZoom.tsx:15]: ditto

File: src/hooks/useAxisWheelZoom.tsx [hunk 2]
```diff
@@ -6,10 +6,31 @@ import {
...
+   * */
   center?: 'pointer' | number;
+  /**
+   * Zoom factor
+   * @default 1
+   * */
   factor?: number;
+  /**
+   * Invert zoom
+   * @default false - Zoom in OnWheel up
```
Comment (stropitek) [inline src/hooks/useAxisWheelZoom.tsx:32]: ditto
Also just put the default value next to `@default` but explanations in the main body (default values won't appear as contextual help when coding in vscode)

File: src/hooks/useAxisZoom.tsx [hunk 3]
```diff
@@ -14,10 +14,30 @@ import { ControllerHookOptions } from './types';
 import { useStartMoveEnd } from './useStartMoveEnd';
 
 export interface UseAxisZoomOptions extends ControllerHookOptions {
+  /**
+   * Zoom direction
+   * @default 'vertical'
+   * */
   direction?: 'horizontal' | 'vertical';
+  /**
+   * Zoom axisId
```
Comment (stropitek) [inline src/hooks/useAxisZoom.tsx:23]: ditto

File: src/hooks/useCrossHair.tsx [hunk 4]
```diff
@@ -9,10 +9,27 @@ import { ControllerHookOptions, DualAxisOptions } from './types';
...
   extends ControllerHookOptions,
     DualAxisOptions {
+  /**
+   * Line color
+   * @default "black"
+   *
+   * */
   color?: CSSProperties['stroke'];
+  /**
+   * Line style
```
Comment (stropitek) [inline src/hooks/useCrossHair.tsx:19]: Try to be a little bit more detailed than what can already be understood from the prop's name.

```suggestion
   * Additional style to apply to the crosshair's lines
```

File: src/hooks/useDrawPath.tsx [hunk 5]
```diff
@@ -11,8 +11,17 @@ export interface UseDrawPathOptions
   extends ControllerHookOptions,
     DualAxisOptions,
     PathOptions {
+  /**
+   * close the path (link the first and last points)
```
Comment (stropitek) [inline src/hooks/useDrawPath.tsx:15]: Start all comments with uppercase
```suggestion
   * Close the path (link the first and last points)
```

File: src/hooks/useDrawPath.tsx [hunk 6]
```diff
@@ -11,8 +11,17 @@ export interface UseDrawPathOptions
...
+   * */
   close?: boolean;
+  /**
+   * callback on drawing
```
Comment (stropitek) [inline src/hooks/useDrawPath.tsx:19]: Might be worth explaining that this is called when the mouse moves

**Review by stropitek — COMMENTED** [review]
I left a few comments in the code but try to review them all with this in mind:

- Don't paraphrase the name of the prop. Try when possible to give additional context.
- Use sentences (starts with an Uppercase and ends with a `.`, no PascalCase / camelCase)

File: src/hooks/useCrossHair.tsx [hunk 7]
```diff
@@ -9,10 +9,27 @@ import { ControllerHookOptions, DualAxisOptions } from './types';
...
+   *
+   * */
+  lineStyle?: CSSProperties;
+  /**
+   * Text Style
```
[the remark this answers is not in the document]
Comment (wadjih-bencheikh18) [reply src/hooks/useCrossHair.tsx]: i didn't understand how can i add svg tag 
for example 'Cross hair svg text style' or 'Cross hair text annotations style'

File: src/hooks/useCrossHair.tsx [hunk 8]
```diff
@@ -9,10 +9,27 @@ import { ControllerHookOptions, DualAxisOptions } from './types';
...
+   * */
+  lineStyle?: CSSProperties;
+  /**
+   * Cross hair text annotation style.
+   *
+   * */
   textStyle?: CSSProperties;
+  /**
+   * transformation to apply to to the crosshair's text .
+   *
```
Comment (stropitek) [inline src/hooks/useCrossHair.tsx:30]:
```suggestion
   * Transformation to apply to to the crosshair's text.
```

File: src/hooks/useCrossHair.tsx [hunk 9]
```diff
@@ -9,10 +9,27 @@ import { ControllerHookOptions, DualAxisOptions } from './types';
...
     DualAxisOptions {
+  /**
+   * Cross hair line color.
+   * @defaultValue "black"
+   *
+   * */
   color?: CSSProperties['stroke'];
+  /**
+   * Additional style to apply to the crosshair's lines.
+   *
```
Comment (stropitek) [inline src/hooks/useCrossHair.tsx:20]:
```suggestion
```

File: src/hooks/useCrossHair.tsx [hunk 10]
```diff
@@ -9,10 +9,27 @@ import { ControllerHookOptions, DualAxisOptions } from './types';
 export interface UseCrossHairOptions
   extends ControllerHookOptions,
     DualAxisOptions {
+  /**
+   * Cross hair line color.
+   * @defaultValue "black"
+   *
```
Comment (stropitek) [inline src/hooks/useCrossHair.tsx:15]:
```suggestion
```

File: src/hooks/useCrossHair.tsx [hunk 11]
```diff
@@ -9,10 +9,27 @@ import { ControllerHookOptions, DualAxisOptions } from './types';
...
+   * */
+  lineStyle?: CSSProperties;
+  /**
+   * Cross hair text annotation style.
+   *
+   * */
   textStyle?: CSSProperties;
+  /**
+   * transformation to apply to to the crosshair's text .
```
Comment (stropitek) [inline src/hooks/useCrossHair.tsx:29]:
```suggestion
   * Transformation to apply to the crosshair's text.
```

File: src/hooks/useCrossHair.tsx [hunk 12]
```diff
@@ -9,10 +9,27 @@ import { ControllerHookOptions, DualAxisOptions } from './types';
...
+   * */
   color?: CSSProperties['stroke'];
+  /**
+   * Additional style to apply to the crosshair's lines.
+   *
+   * */
+  lineStyle?: CSSProperties;
+  /**
+   * Cross hair text annotation style.
+   *
```
Comment (stropitek) [inline src/hooks/useCrossHair.tsx:25]:
```suggestion
```

File: src/hooks/useAxisZoom.tsx [hunk 13]
```diff
@@ -14,10 +14,28 @@ import { ControllerHookOptions } from './types';
...
 
 export interface UseAxisZoomOptions extends ControllerHookOptions {
+  /**
+   * The zoom direction to use on wheel events.
+   * @defaultValue 'vertical'
+   * */
   direction?: 'horizontal' | 'vertical';
+  /**
+   * Id of Axis to apply zoom.
+   * Default value is 'x' when direction is horizontal and 'y' if direction is vertical
```
Comment (stropitek) [inline src/hooks/useAxisZoom.tsx:24]:
```suggestion
   * Default value is 'x' when direction is horizontal and 'y' if direction is vertical.
```

**Comment by stropitek** [comment]
LGTM after the typos are fixed

**Comment by wadjih-bencheikh18** [comment]
@stropitek typos fixed, thanks

**Pull request merged**